### PR TITLE
Properly switching window focus after close of window

### PR DESCRIPTION
**Pull request by jimevans**
The test_handle_prompt_accept test for closing a window tests to see if an
alert was handled. It does so by calling the Get Alert Text end point.
However, the Get Alert Text end point requires an active, valid top-level
browsing context to have the focus. The test now switches focus back to
the previous window before checking that an alert was handled. In between
checking the alert, confirm, and prompt cases, it recreates the new window
required so it can be closed.

### CODE REVIEW

File: webdriver/tests/close_window/user_prompts.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from tests.support.asserts import assert_error, assert_dialog_handled
 from tests.support.fixtures import create_dialog, create_window
 from tests.support.inline import inline
-
+from time import sleep
```
Comment (andreastt) [inline webdriver/tests/close_window/user_prompts.py]: Debug leftover?

Comment (whimboo) [inline webdriver/tests/close_window/user_prompts.py]: Please remove the line which you probably used when debugging the code.

**Review by andreastt — DISMISSED** [review]

**Comment by andreastt** [comment]
@whmboo: Can you take a closer look at this?

**Comment by andreastt** [comment]
@whimboo Can you take a closer look at this?

File: webdriver/tests/close_window/user_prompts.py [hunk 2]
```diff
@@ -24,23 +24,33 @@ def test_handle_prompt_ignore():
...
+    original_handle = session.window_handle
 
+    session.window_handle = create_window(session)()
     session.url = inline("<title>WD doc title</title>")
 
     create_dialog(session)("alert", text="dismiss #1", result_var="dismiss1")
     response = close(session)
     assert response.status == 200
+    session.window_handle = original_handle
     assert_dialog_handled(session, "dismiss #1")
```
Comment (whimboo) [inline webdriver/tests/close_window/user_prompts.py]: We want to make sure that the dialog on the newly opened window has been handled and not on the original window, right? In that case resetting the `window_handle` should come after the assertion.

**Review by whimboo — CHANGES_REQUESTED** [review]

File: webdriver/tests/close_window/user_prompts.py (hunk 2 above)
Comment (whimboo) [reply webdriver/tests/close_window/user_prompts.py]: Hm, given that the dialog should have been accepted by the `close` command, the window is gone. Does it make sense to have the check at all for this command?

File: webdriver/tests/close_window/user_prompts.py (hunk 1 above)
Comment (jimevans) [reply webdriver/tests/close_window/user_prompts.py]: Yep. Will do.

Comment (jimevans) [reply webdriver/tests/close_window/user_prompts.py]: Good catch. Fixing.

File: webdriver/tests/close_window/user_prompts.py (hunk 2 above)
Comment (jimevans) [reply webdriver/tests/close_window/user_prompts.py]: I suppose it makes sense that we assert that no dialog is present, which is what assert_dialog_handled does. However, the call to Get Alert Text (which is what assert_dialog_handled uses to validate whether an alert is present) requires a valid, open top-level browsing context, so the reset of the current window handle must come first.

Comment (whimboo) [reply webdriver/tests/close_window/user_prompts.py]: So lets have a look at the spec for Switch To Window:

> If there is an active user prompt, that prevents the focussing of another top-level browsing context, return error with error code unexpected alert open. 

So similar to other commands switching to a different window will also accept the dialog in this test. So using `assert_dialog_handled` afterward would be a no-op. Best would still be to check that the newly created window is gone, and if not make the dialog check.

**Review by whimboo — APPROVED** [review]
I think this is fine for now. Thanks.

**Review by andreastt — APPROVED** [review]

**Pull request merged**